_books/ion-1-1/src/binary/values/struct.md

```diff
@@ -36,111 +41,113 @@ D6 15 61 01 17 61 02
+Structs may switch between SID Mode and FlexSym Mode by placing a mode-switch opcode (`0xEE`) in the _value_ position of a struct.
+This causes the prior field name to be discarded (just like a `NOP`), and switches the struct from FlexSym Mode to SID Mode or from SID Mode to FlexSym Mode.
 
-Unlike lists and S-expressions, structs cannot use opcode `0xF0` by itself to indicate the end of the delimited
-container. This is because `0xF0` is a valid `FlexSym` (a symbol with 16 bytes of inline text). To close the delimited
-struct, the writer emits a `0x01` byte (a `FlexSym` escape) followed by the opcode `0xF0`.
+Mode-switching works with both prefixed and delimited structs. It costs 2 bytes to switch modes (one for the discarded field name, and one for the mode-switch opcode).
+It is possible, but not recommended, to switch back and forth between modes.
 
-#### Delimited encoding of the empty struct (`{}`)
-```
-┌─── Opcode 0xF3 indicates the beginning of a delimited struct
-│  ┌─── FlexSym escape code 0 (0x01): an opcode follows
-│  │  ┌─── Opcode 0xF0 indicates the end of the most
-│  │  │    recently opened delimited container
-F3 01 F0
-```
+In **SID Mode**, each field name is a `FlexUInt` which is the SID.
+In **FlexSym Mode**, each field name is a `FlexSym`.
 
 > [!NOTE]
-> It is much more compact to write `0xD0`—the [empty length-prefixed struct](#length-prefixed-encoding-of-an-empty-struct-).
+> The ability to switch modes exists to allow writer implementations to start in the more compact SID mode and then switch to FlexSym mode if needed without having to backtrack and rewrite all prior fields in the struct. If you know ahead of time that you will have a mix of inline text and SIDs, you should usually prefer to start a struct in FlexSym mode rather than switching to SID mode later.
 
-#### Delimited encoding of `{"foo": 1, $11: 2}`
+
+##### Switching to FlexSym Mode while encoding `{$10: 1, foo: 2, $11: 3}`
+In this example, the writer switches to FlexSym Mode before encoding `foo` so it can write the UTF-8 bytes inline.
 ```
-┌─── Opcode 0xF3 indicates the beginning of a delimited struct
-│
-│  ┌─ FlexSym -3     ┌─ FlexSym: 11 ($11)
-│  │                 │        ┌─── FlexSym escape code 0 (0x01): an opcode follows
-│  │                 │        │  ┌─── Opcode 0xF0 indicates the end of the most
-│  │   f  o  o       │        │  │    recently opened delimited container
-F3 FB 66 6F 6F 61 01 17 61 02 01 F0
-      └──┬───┘ └─┬─┘    └─┬─┘
-      3 UTF-8    1        2
-       bytes
+┌──── An opcode in the range 0xD0-0xDF indicates a length-prefixed struct
+│  ┌─── Field name: FlexUInt 10 ($10) [SID Mode]
+│  │        ┌─── Throwaway field name
+│  │        │  ┌─── Mode switch opcode 0xEE (switches to FlexSym Mode)
+│  │        │  │  ┌─── FlexSym: -4 (3 UTF-8 bytes follow)
+│  │        │  │  │           ┌─── Field name: FlexSym 11 ($11) [FlexSym Mode]
```

Comment:
The 17 this points to appears superfluous. `2` is preceded by inline `foo` as its field name.